react/src/app.jsx

```diff
@@ -0,0 +1,60 @@
+import {useState} from 'preact/hooks';
+import {Component, createRef} from 'preact';
+import wonderlandLogo from './assets/wonderland-horizontal-reversed.svg';
+import {initializeWonderland} from 'wonderland-react-example';
+import './app.css';
+
+class WonderlandApp extends Component {
+    componentDidMount() {
+        initializeWonderland().then((engine) => this.setState({engine}));
+    }
+
+    render() {
+        return <canvas id="canvas"></canvas>;
+    }
+}
+
+function toggleColor(engine) {
+    const cube = engine.scene.findByName('Cube')[0];
+    const mesh = cube.getComponent('mesh');
+    mesh.material.diffuseColor = [Math.random(), Math.random(), Math.random(), 1.0];
+    return engine;
+}
+
+export class App extends Component {
+    constructor(props) {
+        super(props);
+        this.WonderlandApp = createRef();
+    }
+
+    render() {
+        return (
+            <>
+                <div>
+                    <a href="https://wonderlandengine.com/showcase" target="_blank">
+                        <img
+                            src={wonderlandLogo}
+                            class="logo wonderland"
+                            alt="Wonderland Engine logo"
+                        />
+                    </a>
+                </div>
+                <h1>React Example</h1>
+                <div class="card">
+                    <p>
+                        <WonderlandApp ref={this.WonderlandApp}></WonderlandApp>
+                    </p>
+                    <button
+                        onClick={() => toggleColor(this.WonderlandApp.current.state.engine)}
```

Comment:
See other comment, manipulating a prop forwarded from this component is maybe better for the example. While both works, it's a common thing to pass down data using props and react to changes